### PR TITLE
Format attribute for widgets, popups and legends [ch60845]

### PR DESCRIPTION
**Pull request by neokore**
Related to https://app.clubhouse.io/cartoteam/story/60845/allow-to-format-attributes

### CODE REVIEW

**Comment by Jesus89** [comment]
It seems that the linter issue is only with the bundle because of the new dependency d3/format. So we must exclude the bundle from the linter check.

File: cartoframes/viz/widgets/histogram_widget.py [hunk 1]
```diff
@@ -1,7 +1,8 @@
 from ..widget import Widget
 
 
-def histogram_widget(value, title=None, description=None, footer=None, read_only=False, buckets=20, weight=1):
+def histogram_widget(value, title=None, description=None, footer=None, read_only=False,
+                     buckets=20, weight=1):
```
Comment (VictorVelarde) [inline cartoframes/viz/widgets/histogram_widget.py:5]: why this line CR into 2 lines?

**Comment by VictorVelarde** [comment]
The python side look good to me. Let's see if we can get travis to pass

File: cartoframes/viz/widgets/histogram_widget.py (hunk 1 above)
Comment (neokore) [reply cartoframes/viz/widgets/histogram_widget.py:5]: This is to avoid too long line warning

**Comment by neokore** [comment]
Pushed changes for Hound and Travis CI.

**Comment by neokore** [comment]
Travis forced to rebuild and now it passes. A :heavy_check_mark: for this?

File: examples/popups/popup_titles_and_format.ipynb [hunk 2]
```diff
@@ -1286,19 +1286,14 @@
     "from cartoframes.viz import Layer, popup_element\n",
     "\n",
     "Layer(\n",
-    "    'countries_africa',\n",
+    "    'spain_populated_places',\n",
+    "    basic_style(size=10),\n",
     "    popup_hover=[\n",
-    "        popup_element('name', 'Country', 'population')\n",
+    "        popup_element('name', 'City'),\n",
+    "        popup_element('pop_max', title='Rounded population', format=',.2r')\n",
```
Comment (VictorVelarde) [inline examples/popups/popup_titles_and_format.ipynb:1293]: LGTM. I'm gonna ping here @manmorjim as he's interested in popup and formatting. It looks like using d3.format on both CF and web-sdk could be nice combination

**Pull request merged**